internal/controllers/topology/cluster/scope/upgradetracker.go

```diff
@@ -222,3 +244,123 @@ func (m *MachineDeploymentUpgradeTracker) DeferredUpgradeNames() []string {
 func (m *MachineDeploymentUpgradeTracker) DeferredUpgrade() bool {
 	return len(m.deferredNames) != 0
 }
+
+// MachinePoolUpgradeTracker holds the current upgrade status and makes upgrade
```

Comment:
Same here. The MD part probably evolved since you wrote this code. Please check/compare

(I'll skip this part for now)

I think the mismatch is mostly rollingout vs upgrading. Should be fine if you re-diff MD vs MP in this file and in state.go